tests/model/SiteTreeTest.php

```diff
@@ -108,6 +109,47 @@ public function testURLGeneration()
     }
 
     /**
+     * Check if reserved URL's are properly appended with a number at top level
+     */
+    public function testDisallowedURLGeneration()
+    {
+        // Expected URL defaults to '-2'
+        $expectedURLs = [
```

Comment:
Good idea, but I think it would be better to do such a thing in all the tests at once, instead of having some of the tests use inline expectation arrays (as `testURLGeneration` on line 88) and others use providers